modules/katello_upload.py

```diff
@@ -162,36 +132,41 @@ def main():
     if has_import_error:
         module.fail_json(msg=import_error_msg)
 
-    server_url = module.params['server_url']
-    username = module.params['username']
-    password = module.params['password']
-    src = module.params['src']
-    repository = module.params['repository']
-    product = module.params['product']
-    organization = module.params['organization']
-    verify_ssl = module.params['verify_ssl']
-
-    server = ServerConfig(
-        url=server_url,
-        auth=(username, password),
-        verify=verify_ssl
-    )
-    ng = NailGun(server, entities, module)
+    entity_dict = dict(
+        [(k, v) for (k, v) in module.params.items() if v is not None])
+
+    server_url = entity_dict.pop('server_url')
+    username = entity_dict.pop('username')
+    password = entity_dict.pop('password')
+    verify_ssl = entity_dict.pop('verify_ssl')
 
-    # Lets make an connection to the server with username and password
     try:
-        org = entities.Organization(server)
-        org.search()
+        create_server(server_url, (username, password), verify_ssl)
     except Exception as e:
         module.fail_json(msg="Failed to connect to Foreman server: %s " % e)
 
-    try:
-        if not module.check_mode:
-            ng.upload(src, repository, product, organization)
-    except Exception as e:
-        module.fail_json(msg=to_native(e))
+    ping_server(module)
+
+    name, version, release, arch = check_output("rpm --queryformat '%%{NAME} %%{VERSION} %%{RELEASE} %%{ARCH}' -qp %s" % entity_dict['src'],
```

Comment:
@sean797 FYI -- This breaks all non-RPM uploads (e.g. files, puppet modules). I think @pcreech is going to submit a fix proposal if you could review when he does.

Reply:
Yes - sure!
Thanks